docs/testing/e2e-plan.md

```diff
@@ -0,0 +1,79 @@
+  Tests should be run against an arbitrary versions of artifacts, defaulting to their latest versions.
+
++ **Timing:**
+  E2E tests should be run on every revision of an "ok to test" PR. This may present some challenges for the current E2E test architecture, as it might require communication between separate AWS pipelines, or failing that, custom code that can communicate with other AWS resources to identify and list available artifacts built in other pipelines.
+
++ **Platform configuration:**
+  A test should be able to exclude specific EKS-A versions, EKS-A providers, cluster node OSes, or K8s versions with which they are incompatible or unsupported. A test will be run on every permutation by default.
+
++ **Configuration review & control:**
+  The test dimensions should be easily viewed, verified, and edited by humans (or reasonable facsimiles thereof). This provides a means of quickly excluding tests categorically for reasons such as hardware outage.
+
+Additionally, it is foreseen that the number of E2E tests that need to be run for each PR will quickly grow, and so it is recommended that these performance goals also be considered:
+
++ **Reusable clusters:** The amount of time required to create and destroy clusters before and after each test will quickly become overwhelming with reasonable amounts of hardware.
+
+## The Path Forward
+
+### Short Term
+
+#### Test Against Current Branch
+
+The current tests will be modified to run against the latest versions of artifacts. This will be accomplished by removing the existing hardcoded registry links from the existing tests, and implementing a means by which URLs to test-run-specific artifacts will be used. These up-to-date artifacts will be used (at a minimum) when installing the EKS-A Curated Packages controller during cluster creation, as well as when said controller acquires its active bundle. The newly-acquired active bundle will likewise contain links to newly-built test artifacts, such that the end-to-end tests will be run against artifacts built using the current git commits under test, and not the hard-coded development or production registries, as is the case now.
+
+Test runs should be accomplished using development ECR repositories (i.e. public ECR), but if or where that isn't the case, credentials for accessing a private ECR would have to be acquired, which could prove a technical risk, as to my knowledge, this isn't presently done.
+
+#### Investigate Daily E2E Run
+
+Running end-to-end tests more frequently is another priority of the short term plan. An investigation into the feasibility of adding a periodic run of the end-to-end tests will be performed. If a periodic end-to-end run is estimated to be within a week's work, then it will be undertaken. The intention is to be notified of end-to-end test failures before a release branch is created, when deadlines are already set and there are possibly many time-sensitive tasks already waiting to be completed.
```

Comment:
I assume these daily E2E runs will be done on the main branch in CP instead of against the latest artifacts right?

Reply:
> I assume these daily E2E runs will be done on the main branch in CP instead of against the latest artifacts right?

I was planning for these to run against main. In the context of a daily run, what would be more appropriate? What do you consider "the latest artifacts" if not those built from main?

Reply:
I do envision a potential scenario where we will be maintaining/patch release branches in which case I imagine we would want to run E2E on in changes in those branches.